proposals/1219-storing-megolm-keys-serverside.md

```diff
@@ -172,6 +172,8 @@ On success, returns a JSON object with keys:
 - `auth_data` (object): Required. Same as in the body parameters for
   `POST /room_keys/version`.
 - `version` (string): Required. The backup version.
+- `hash` (string): Required. A hash value representing stored keys.
```

Comment:
We need more details about what exactly this "hash" is.  Is it, for example, a cryptographic hash of some specific string?  Or is it more like [ETag in HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag), where it's just an opaque string that the server generates, and the client just checks for equality?

Reply:
Yes. This is like more `Etag` except this is the client that checks for equality. I used a generic term like `hash` because I had few ideas about naming. The other word coming into my mind was `token` but in matrix, `token` is associated to some kind of pagination or streaming we do not have here (unfortunately).

Reply:
The thing is that something like an etag in itself, based on all the keys in the backup, wouldn't really be useful as it would change whenever the device itself uploaded a key - the device only cares about keys it doesn't have.

Reply:
> the device only cares about keys it doesn't have.

True. This hash is here only to solve the case where the count of keys in the backup is the same as on the client. How can we be sure keys are the same on both side?
The final goal is to know when to display a restore button to the end user.